File: 7labs/calc.ts
class Calculation {
    private a: number;
    private c: number;

    constructor(a: number, c: number) {
        this.a = a;
        this.c = c;
    }

    calculateExpression(): number {
        if (this.c === -this.a) {
            throw new Error("Division by zero is not allowed.");
        }

        return Math.log((2 * this.c - this.a) + 4 - 152) / (this.a / this.c);
    }
}

const calculations: Calculation[] = [
    new Calculation(4, 2),
    new Calculation(5, -5),
    new Calculation(3, 3)
];

calculations.forEach((calc, index) => {
    try {
        console.log(`Result for calculation ${index + 1}: ${calc.calculateExpression()}`);
    } catch (error) {
        console.error(`Error for calculation ${index + 1}: ${(error as Error).message}`);
    }
});

File: 7labs/calc.cs
using System;
using System.Collections.Generic;

namespace ExceptionHandlingExample
{
    class Calculation
    {
        public double A { get; set; }
        public double C { get; set; }

        public Calculation(double a, double c)
        {
            A = a;
            C = c;
        }

        public double CalculateExpression()
        {
            if (C == -A)
                throw new ArgumentException("Division by zero is not allowed.");

            return Math.Log((2 * C - A) + 4 - 152) / (A / C);
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            List<Calculation> calculations = new List<Calculation>
            {
                new Calculation(4, 2),
                new Calculation(5, -5),
                new Calculation(3, 3)
            };

            foreach (var calc in calculations)
            {
                try
                {
                    Console.WriteLine($"Result: {calc.CalculateExpression()}");
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"Error: {ex.Message}");
                }
            }
        }
    }
}
